examples/measure/rs-measure.cpp

```diff
@@ -0,0 +1,553 @@
+                    for (auto&& v : n)
+                    {
+                        // If this pixel was not yet visited, initialize
+                        // its distance as +INF
+                        if (dist.find(v) == dist.end()) dist[v] = INFINITY;
+
+                        // Calculate distance in 3D between the two neighboring pixels
+                        auto d = dist_3d(intrinsics, depth, u, v);
+                        // Calculate total distance from source
+                        auto total_dist = dist[u] + d;
+
+                        // If we encounter a potential improvement,
+                        if (dist[v] > total_dist)
+                        {
+                            // Update parent and distance
+                            parent[v] = u;
+                            dist[v] = total_dist;
+                            // And re-visit that pixel by re-introducing it to the queue
+                            q.emplace(total_dist, v);
+                        }
+                    }
+                }
+
+                {
+                    // Write the shortest-path to the path variable
+                    std::lock_guard<std::mutex> lock(path_mutex);
+                    total_dist = dist[trg];
+                    path.clear();
+                    // Iterate until encounter token special pixel
+                    while (trg != token)
+                    {
+                        // Handle the case we didn't find a path
+                        if (trg == parent[trg]) break;
+
+                        path.emplace_back(trg);
+                        trg = parent[trg];
+                    }
+                }
+            }
+        }
+    });
+
+    while(app) // Application still alive?
+    {
+        // Fetch the latest available post-processed frameset
+        static rs2::frameset current_frameset;
+        postprocessed_frames.poll_for_frame(&current_frameset);
+
+        if (current_frameset)
+        {
+            auto depth = current_frameset.get_depth_frame();
+            auto color = current_frameset.get_color_frame();
+
+            glEnable(GL_BLEND);
+            // Use the Alpha channel for blending
+            glBlendFunc(GL_SRC_ALPHA, GL_ONE_MINUS_SRC_ALPHA);
+
+            // First render the colorized depth image
+            depth_image.render(depth, { 0, 0, app.width(), app.height() });
+            // Next, set global alpha for the color image to 90%
+            // (to make it slightly translucent)
+            //glColor4f(1.f, 1.f, 1.f, 0.9f);
+            // Render the color frame (since we have selected RGBA format
+            // pixels out of FOV will appear transparent)
+            color_image.render(color, { 0, 0, app.width(), app.height() });
+
+            {
+                // Take the lock, to make sure the path is not modified
+                // while we are rendering it
+                std::lock_guard<std::mutex> lock(path_mutex);
```

Comment:
This new scope could be refactored out if executing my suggestion in the previous comment

Reply:
If you're not going to refactor the threads, I think performance could be improved if you take the mutex here, swap the `path` with a local vector, and unlock the mutex before rendering the path. (This looks safe since you call `path.clear()` in the thread right before refilling it